spec/requests/api/suppliers_controller_locations_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::SuppliersController do
+  let(:access_token) { 'spoofed-token' }
+  let(:headers) { { 'CONTENT_TYPE': content_type, 'Authorization': "Bearer #{access_token}" } }
+  let(:content_type) { ApiController::CONTENT_TYPE }
+  let(:response_json) { JSON.parse(response.body) }
+
+  describe 'GET /suppliers/:supplier_id/locations' do
+    let(:supplier) { create(:supplier) }
+
+    it 'returns the locations' do
+      location1 = create(:location, suppliers: [supplier])
+      location2 = create(:location, suppliers: [supplier])
+      create_list(:move, 2, :requested, supplier: supplier, from_location: location1) # to tests that duplicates locations are not shown
+      create(:move, :requested, supplier: supplier, from_location: location2)
+
+      get "/api/v1/suppliers/#{supplier.id}/locations", headers: headers, params: {}
+
+      response_location_keys = response_json['data']
+                                 .select { |e| e['type'] == 'locations' }
+                                 .map { |l| l['attributes']['key'] }
+
+      expect(response).to have_http_status(:success)
+      expect(response_location_keys).to match_array([location1.key, location2.key])
```

Comment:
Would prefer to split this into two separate tests ideally - also could this use the standard approach:

```ruby
      it_behaves_like 'an endpoint that responds with success 200'
```

Would be nice to move the request out into a subject line so it can be called as needed too.

e.g.

```ruby
subject(:get_supplier_locations) { get "/api/v1/suppliers/#{supplier_id}/locations", headers: headers, params: {} }

it_behaves_like 'an endpoint that responds with success 200' do
  get_supplier_locations
end

Reply:
nice one, yes I'agree, let's make this test more "standard"